src/utils.py

```diff
@@ -158,14 +158,58 @@ def run_command(command, env=None, check=True):
         debug_log("::endgroup::")
 
 
-def error_exit(branch_name: str):
-    """Cleans up a branch (if provided), sends telemetry, and exits with code 1."""
+def error_exit(remediation_id: str, failure_code: Optional[str] = None):
```

Comment:
Maybe you could have an additional failure message `failure_message: Optional[str] = None` passed here for more info to send?